src/components/TableView/TableView.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { FC, useState, memo } from "react";
+
+import Avatar from "@material-ui/core/Avatar";
+import Checkbox from "@material-ui/core/Checkbox";
+import Paper from "@material-ui/core/Paper";
+import Table, { TableProps } from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TablePagination from "@material-ui/core/TablePagination";
+import TableRow from "@material-ui/core/TableRow";
+
+import TableToolbar from "./TableToolbar";
+import { useStyles } from "./TableView.styles";
+import { ColumnsOptionProps } from "./TableView.types";
+import { Rows } from "./TableView.types.example";
+
+interface TableViewProps {
+  title?: string;
+  columns: (string | ColumnsOptionProps)[];
+  rows: Rows[];
+  tableProps?: TableProps;
+  // paginationProps starts here
+  withPagination?: boolean;
+  onPaginationChange?: (newPage: number) => void;
+  rowsPerPage?: number;
+  count?: number;
+  // paginationProps ends here
+  withSearch?: boolean;
+  withCheckbox?: boolean;
+}
+
+const TableView: FC<TableViewProps> = ({
+  title = "Table Example",
+  columns,
+  rows,
+  tableProps,
+  withPagination = false,
+  onPaginationChange = () => {
+    return null;
+  },
+  rowsPerPage = 0,
+  count = 0,
+  withSearch = false,
+  withCheckbox = false,
+}) => {
+  const [page, setPage] = useState(0);
+  const [selected, setSelected] = useState<string[]>([]);
+
+  const classes = useStyles();
+
+  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const newSelecteds = rows.map(n => n.email);
```

Comment:
This `TableView` component must be a common component and will be used with multiple purposes, so I think so shouldn't  use a specific attribute to mark a row is selected or not. In my opinion, we can add 1 more props (Ex: rowKey) into this component. This prop can be used to mark selected rows and used as a key for rows loop rendering

Reply:
Good point, thanks.
I just fixed it with the next commit: ae45219